warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/handler/facet/FacetHandlerTest.java

```diff
@@ -0,0 +1,595 @@
+        
+        evaluateSingleEventResults(keysByTable, expectedFacets, expectedFacetMetadata);
+    }
+    
+    @Test
+    public void testMultipleEvents() {
+        ExtendedDataTypeHandler<Text,BulkIngestKey,Value> handler = new FacetHandler<>();
+        handler.setup(ctx);
+        
+        helper.setup(ctx.getConfiguration());
+        
+        Collection<Multimap<String,NormalizedContentInterface>> items = TestData.getDataItems();
+        int size = items.size();
+        
+        setupTaskAttemptContext();
+        items.forEach(f -> processEvent(event, f, handler));
+        Multimap<String,FacetResult> keysByTable = collectResults();
+        
+        Object2IntMap<String> expectedFacetKeyCounts = TestData.getExpectedFacetCounts();
+        evaluateMultipleEventResults(keysByTable, size, expectedFacetKeyCounts);
+    }
+    
+    @Test
+    public void testMultiValueHashing() {
+        
+        ctx.getConfiguration().set("test.facet.hash.threshold", "2");
+        
+        ExtendedDataTypeHandler<Text,BulkIngestKey,Value> handler = new FacetHandler<>();
+        handler.setup(ctx);
+        
+        Collection<Multimap<String,NormalizedContentInterface>> items = TestData.generateMultivaluedColorData();
+        
+        setupTaskAttemptContext();
+        items.forEach(f -> processEvent(event, f, handler));
+        Multimap<String,FacetResult> keysByTable = collectResults();
+        
+        evaluateMultiValueResults(keysByTable, TestData.getExpectedFacetHashes());
+    }
+    
+    StandaloneTaskAttemptContext<Text,RawRecordContainerImpl,BulkIngestKey,Value> sctx;
+    CachingContextWriter contextWriter;
+    
+    private void setupTaskAttemptContext() {
+        sctx = new StandaloneTaskAttemptContext<>(ctx.getConfiguration(), new StandaloneStatusReporter());
+        contextWriter = new CachingContextWriter();
+        try {
+            contextWriter.setup(sctx.getConfiguration(), false);
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException("Error setting up context writer", e);
+        }
+    }
+    
+    /**
+     * Process a single event using the handler
+     *
+     * @param event
+     *            the event we're processing
+     * @param eventFields
+     *            the fields from the event to process
+     * @param handler
+     *            the handler to do the procesing
+     */
+    private void processEvent(RawRecordContainer event, Multimap<String,NormalizedContentInterface> eventFields,
+                    ExtendedDataTypeHandler<Text,BulkIngestKey,Value> handler) {
+        
+        assertNotNull("Event was null", event);
+        
+        try {
+            handler.process(null, event, eventFields, sctx, contextWriter);
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException("Error setting up context writer", e);
```

Comment:
This error message seems like it was copied from elsewhere in the test & should probably say that there was an error in processing

Reply:
Resolved